pkg/webhooks/github/actions/issues_handler.go

```diff
@@ -126,23 +128,58 @@ func (r *IssuesAction) buildForkPR(ctx context.Context, p *IssuesActionParams) e
 	}
 	targetRepoURL.User = url.UserPassword("x-access-token", token)
 
-	headRef := *pullRequest.Head.Ref
-	commitMessage := "Triggering fork build approved by maintainer"
-	err = git.PushToRemote(*pullRequest.Head.Repo.CloneURL, headRef, targetRepoURL.String(), "fork-build/"+headRef, commitMessage)
+	var (
+		commitMessage   = "Triggering fork build approved by maintainer"
+		headRef         = *pullRequest.Head.Ref
+		prNumber        = strconv.Itoa(*pullRequest.Number)
+		forkBuildBranch = "fork-build/" + prNumber
+		forkPrTitle     = "Fork build for #" + prNumber
+	)
+
+	err = git.PushToRemote(*pullRequest.Head.Repo.CloneURL, headRef, targetRepoURL.String(), forkBuildBranch, commitMessage)
 	if err != nil {
 		return fmt.Errorf("error pushing to target remote repository %w", err)
 	}
 
-	r.logger.Info("triggered fork build action by pushing to fork-build branch", "source-repo", p.RepositoryName, "branch", headRef)
+	forkPr, _, err := r.client.GetV3Client().PullRequests.Create(ctx, r.client.Organization(), p.RepositoryName, &github.NewPullRequest{
+		Title:               github.String(forkPrTitle),
+		Head:                github.String(forkBuildBranch),
+		Base:                github.String(*pullRequest.Base.Ref),
+		Body:                github.String("Fork build for #" + prNumber + " triggered by @" + p.User),
+		MaintainerCanModify: github.Bool(true),
+	})
+	if err != nil {
+		if !strings.Contains(err.Error(), "A pull request already exists") {
+			return err
+		}
+	}
+
+	// and immediately close this PR again, it's just for building...
+	forkPr.State = github.String("closed")
+
+	_, _, err = r.client.GetV3Client().PullRequests.Edit(ctx, r.client.Organization(), p.RepositoryName, *forkPr.Number, forkPr)
+	if err != nil {
+		return err
+	}
+
+	r.logger.Info("triggered fork build action by pushing to fork-build branch", "source-repo", p.RepositoryName, "branch", forkBuildBranch, "pull-request-url", forkPr.GetURL())
 
 	_, _, err = r.client.GetV3Client().Reactions.CreateIssueCommentReaction(ctx, r.client.Organization(), p.RepositoryName, p.CommentID, "rocket")
 	if err != nil {
 		return fmt.Errorf("error creating issue comment reaction %w", err)
 	}
 
-	err = git.DeleteBranch(targetRepoURL.String(), "fork-build/"+headRef)
+	_, _, err = r.client.GetV3Client().Issues.CreateComment(
```

Comment:
I think it is fine to comment, but not necessarily required as GitHub will display an event on the source PR as the fork PR references it.

Would it make sense to delete all previous `Triggered build for this fork at #%d`-PRs? When testing a few times, this might clutter the PR.

Reply:
That's true, the comment should not be required. But how to delete old PRs? Isn't this the same as closing them?

Reply:
Oh, I think I meant deleting the comments:
> Would it make sense to delete all previous `Triggered build for this fork at #%d`-**comments**? When testing a few times, this might clutter the PR.

Reply:
Because the stuff is pretty hard to test, I just deleted the comment creation because the PR will appear in the PR thread anyway due to the reference in the fork pr body.